articles/making-api-requests-using-retrofit-android/index.md

```diff
@@ -0,0 +1,204 @@
+This article will illustrate how to make API requests using Retrofit. We will create a simple application that shows a list of Marvel characters from an API.
+
+### Introduction
+Networking is a crucial factor in mobile development. Most, if not all mobile applications incorporate networking. Applications are either sending or receiving information. Initially, developers did networking on the `main thread`. This made applications less user-friendly since screens would "freeze". Networking on the main thread stopped after the Honeycomb version was released. Google then developed Volley in 2013. You can read my article on Volley [here](https://www.section.io/engineering-education/making-api-requests-using-volley-android/). Volley offered something better: It was faster, better functionality, simpler syntax, etc. Still, there was more room for growth when it came to networking. Square introduced Retrofit. **Retrofit** is a type-safe HTTP Networking library used for Android and Java. Retrofit was even better since it was super fast, better functionality, and even simpler syntax.
+Most developers have then switched to using Retrofit to make API requests.
+
+### Uses of Retrofit.
+Retrofit is used to perform the following tasks:
+- It manages the process of receiving, sending, and creating HTTP requests and responses.
+- It alternates IP addresses if there is a connection to a web service failure.
+- It caches responses to avoid sending duplicate requests.
+- Retrofit pools connections to reduce latency.
+- Retrofit resolves issues before sending an error and crashing the app.
+
+
+### Advantages of Retrofit.
+- It is very fast.
+- It enables direct communication with the web service.
+- It is easy to use and understand.
+- It supports request cancellation.
+- It supports post requests and multipart uploads.
+- It supports both synchronous and asynchronous network requests.
+- Supports dynamic URLs.
+- Supports convertors.
+
+
+### Disadvantages of Retrofit.
+- It does not support caching.
```

Comment:
You mentioned that retrofit caches responses yet you say that it does not support caching. Please explain what type  of caching is not supported.

Reply:
Thank you for the observation. This was a mistake. The line wasn't meant to be there since Retrofit does actually carry out caching.
Thank you for pointing it out.